src/Symfony/Component/VarDumper/Dumper/AbstractDumper.php

```diff
@@ -68,6 +73,43 @@ public function setOutput($output)
     }
 
     /**
+     * Sets the default character encoding to use for non-UTF8 strings.
+     *
+     * @param string $charset The default character encoding to use for non-UTF8 strings.
+     *
+     * @return The previous charset.
+     */
+    public function setCharset($charset)
+    {
+        $prev = $this->charset;
+        $this->charsetConverter = 'fallback';
+
+        $charset = strtoupper($charset);
+        $charset = null === $charset || 'UTF-8' === $charset || 'UTF8' === $charset ? 'CP1252' : $charset;
```

Comment:
Because the defined charset is to be used to convert **non-utf8** string *to* utf8.
So it makes no sense to say that non-UTF8 string should considered as UTF-8 when converting them to UTF-8.
Well, it makes one sense, which is: I don't expect non-utf8 strings. In that case, if we still encourter one (binary string), then we choose cp1252.